src/CssVariables.js

```diff
@@ -60,11 +60,13 @@ CssVariables.defaultProps = {
  * @prop {boolean} [elevations]
  */
 CssVariables.propTypes = {
+    /* eslint-disable react/no-unused-prop-types */
```

Comment:
Disabled the rule here because these props are used, but just not directly.

Reply:
Suppose we'd change the implementation of `CssVariables` a bit:
```javascript
const CssVariables =({ colors, theme, layers, spacers, elevations }) => {
    const variables = Object.keys({ colors, theme, layers, spacers, elevations })
    // ....
```
This is quite verbose, but it does prevent potential problems: in the current implementation, whenever a prop is added of a non-existent theme key, the component throws an error, i.e.
```html
<CssVariables colors theme layers spacers elevations someThingElse />
```
This will result in the following error:
`Uncaught TypeError: Cannot convert undefined or null to object`

Reply:
Yeah we decided to leave that behaviour in to fail clearly when there are spelling errors in the props. Otherwise you won't be explicitly notified (there'll be a prop-type warning in development, but that's it). We could do it this way as well, I'm fine with both approaches.

I don't think it's a problem to have to disable a rule by the way, as long as it doesn't happen 
that often. It cannot catch certain prop usage, but weighed against the benefit of being warned about unused props is worth it for me. On 124 files having to add the disable to 1 file isn't too bad in my opinion.

Reply:
> Yeah we decided to leave that behaviour in to fail clearly

IMO this isn't that clear... The error is quite cryptic. I would have preferred to have seen `PropTypes.exact` being used to give a clear message, and the code not failing.

Reply:
BTW: sorry, I should've gotten involved in the `CssVariables` PR instead of suggesting changes here.

Reply:
> IMO this isn't that clear..

I agree, I also don't think it's that clear. We discussed it here: https://github.com/dhis2/ui-core/pull/421#discussion_r324646233

If we only want warnings in development we could go with your suggestion. And as you say, we could use propTypes.exact to catch non-existent props. That seems good to me.

It will however fail silently in production then. If we also want to fail clearly in production I think we should throw a clear error on non-existent props.

Reply:
I'd prefer using your suggestion (propTypes.exact and destructuring) and failing clearly in production as well. It doesn't add a lot of extra code or complicated logic and I like that better than failing silently. At that point you can argue that the prop-types don't add much, but that's another discussion, I'm all for optimizing that as well if there's a nicer solution.

Reply:
How about explicitly throwing an error? Then we'd have control over the error message as well

Reply:
I'll work on a PR. Let's discuss further there.